consensus/src/sync/light/validity_window.rs

```diff
@@ -51,17 +51,71 @@ impl<TNetwork: Network> LightMacroSync<TNetwork> {
         election_in_window: bool,
     ) {
         if self.validity_requests.is_none() {
-            // In this case we are going to start a new request
-            let epoch_number = Policy::epoch_at(verifier_block_number);
-            let chunk_index = 0;
+            // By default we set the parameters assuming we are starting from stratch
+            let mut epoch_number = Policy::epoch_at(verifier_block_number);
+            let mut chunk_index = 0;
+            let mut root_hash = expected_root.clone();
+            let mut last_chunk_items: Option<usize> = None;
+            let mut verifier_block_number = verifier_block_number;
+            let mut election_in_window = election_in_window;
+
+            // First we need to check if we already have items in the history store, and request only the remaining portion (if any)
+            match &self.blockchain {
+                BlockchainProxy::Full(blockchain) => {
+                    let blockchain_wr = blockchain.read();
+
+                    let (first_bn, last_bn) = blockchain_wr.history_store.history_store_range(None);
+
+                    // This means we already know the first epoch so we need to request the missing items from the current epoch
```

Comment:
What does the first epoch have to do with that? I assume you mean the epoch in which the start of the window is. Reading this and trying to understand it I do not necessarily know yet there is at most 2 relevant epochs.

Reply:
Yes, the comment is misleading. I will update it